File: source_funcs/go_transcoder/MINIMUM_NUMBER_SUBSETS_DISTINCT_ELEMENTS/MINIMUM_NUMBER_SUBSETS_DISTINCT_ELEMENTS.go
package main

import "sort"

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

func f_gold(ar []int, n int) int {
	var res int = 0
	sort.Ints(ar)
	for i := int(0); i < n; i++ {
		var count int = 1
		for ; i < n-1; i++ {
			if ar[i] == ar[i+1] {
				count++
			} else {
				break
			}
		}
		res = max(res, count)
	}
	return res
}

File: rust_funcs/cpp_transcoder/MINIMUM_NUMBER_SUBSETS_DISTINCT_ELEMENTS/MINIMUM_NUMBER_SUBSETS_DISTINCT_ELEMENTS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(ar: &mut [i32], n: i32) -> i32 {
   let mut res = 0;
   use std::cmp::Reverse;
   let mut i = 0;
   while i < n as usize - 1 {
       let mut count = 1;
       while i < n as usize - 1 && ar[i] == ar[i+1] {
           count += 1;
           i += 1;
       }  
       res = res.max(count);
       i += 1;  
   }
   res  
}
